conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,256 @@ class CMakeFindPackageGenerator(Generator):
         endif()
         """)
 
+    find_components_tpl = Template(textwrap.dedent("""\
+        ########## MACROS ###########################################################################
+        #############################################################################################
+        {{ conan_message }}
+        {{ conan_find_apple_frameworks }}
+        {{ conan_package_library_targets }}
+
+        ########### FOUND PACKAGE ###################################################################
+        #############################################################################################
+
+        include(FindPackageHandleStandardArgs)
+
+        conan_message(STATUS "Conan: Using autogenerated Find{{ pkg_name }}.cmake")
+        set({{ pkg_name }}_FOUND 1)
```

Comment:
This should be populated by the `find_package_handle_standard_args` call if all the arguments are valid. So this is not needed, and the `include` and the call should be the last two lines of the `FindXXXX.cmake` file. Probably we want to add other variables to this check, but we can leave it for the future.

Reply:
Not needed now, we can wait for the refactor